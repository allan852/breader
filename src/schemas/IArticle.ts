export default interface IArtilce {
    author: string
    // categories?: string[]
    // comments?: string
    created_at?: number
    date: string
    feed_id?: number
    time: string
    description?: string
    feed_title?: string
    // enclosures?: InterfaceFeedEnclosure[]
    guid: string
    // image?: InterfaceFeedImage
    id?: number
    index?: number
    is_starred?: number
    is_unread?: number
    is_dayfirst?: boolean
    link: string
    // origlink?: string
    // permalink?: string
    // pubdate: string
    // source?: InterfaceFeedSource
    summary: string
    title: string
    updated_at?: number
};
