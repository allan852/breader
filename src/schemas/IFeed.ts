export default interface IFeed {
    id?: number
    title?: string
    link: string
    url: string
    date?: Date
    date_time?: string
    etag?: string
    description?: string
    favicon?: string
    summary?: string
    deleted_at?: number
}
