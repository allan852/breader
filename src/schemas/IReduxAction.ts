export interface IReduxAction<P = any> {
    payload: P,
    type: string,
}
