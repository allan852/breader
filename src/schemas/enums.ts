export enum ELogicError {
  FEEDPARSER_FETCH_ERROR = 'FEEDPARSER_FETCH_ERROR',
  FEEDPARSER_NOT_FOUND = 'FEEDPARSER_NOT_FOUND',
  FEEDPARSER_WRONG_URL = 'FEEDPARSER_WRONG_URL',
  POUCHDB_EXISTS = 'POUCHDB_EXISTS',
  UNKNOWN = 'UNKNOWN',
}

export enum EMenuKey {
  ALL_ITEMS = 'ALL_ITEMS',
  STARRED_ITEMS = 'STARRED_ITEMS',
  UNREAD_ITEMS = 'UNREAD_ITEMS',
}

export enum EArticleFilter {
  STARRED = 'STARRED',
  UNREAD = 'UNREAD',
  ALL = 'ALL',
}
