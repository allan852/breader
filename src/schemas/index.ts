export * from './enums'
export * from './IArticle'
export * from './IFeed'
export * from './IReduxAction'
export * from './IReduxAsyncAction'
export * from './IReduxState'
