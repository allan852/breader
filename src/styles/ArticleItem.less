@import './Values.less';
.article-item {
    cursor: pointer;
    color: @black-text-color-dark;
    display: flex;
    padding: 4px 8px 4px 8px;
    .item-sider {
        width: 26px;
        flex: 0 0 26px;
    }
    .item-main {
        flex: 1 1 auto;
    }
    .item-header {
        display: block;
        height: 18px;
        line-height: 18px;
        font-size: 12px;
    }
    .item-header-left {
        float: left;
    }
    .item-header-right {
        float: right;
    }
    .item-content {
        font-size: 13px;
        font-weight: 500;
        line-height: 20px;
    }
    &.item-is-unread .item-content {
        color: @black-text-color-light;
    }
    &.item-is-selected {
        background-color: @list-selected-color;
    }
    .item-footer {
        width: 220px;
        overflow: hidden;
        text-overflow: ellipsis;
        white-space: nowrap;
        line-height: 18px;
        font-size: 12px;
    }
}