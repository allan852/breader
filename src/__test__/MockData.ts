import { IntlProvider } from 'react-intl'
import { messages } from '../locales'
import { IArticle, IFeed } from '../schemas'

export const intlProviderProps: IntlProvider.Props = {
  locale: 'en-US',
  messages: messages['en-US'],
}

const intlProvider = new IntlProvider(intlProviderProps, {})
export const intl = intlProvider.getChildContext().intl

export const feed: IFeed = {
  _id: 'http://www.ruanyifeng.com/blog/atom.xml',
  _rev: '5-55772bf2b4d4b74608a60ac3480334e4',
  author: 'ruanyifeng',
  categories: [],
  createTime: 1563686468437,
  deleteTime: 0,
  description: "Ruan YiFeng's Blog",
  etag: '"11d86-58e2cc1eab031"',
  favicon: 'http://www.ruanyifeng.com/favicon.ico',
  generator: 'Movable Type 5.2.2 (http://www.sixapart.com/movabletype/)',
  language: '',
  link: 'http://www.ruanyifeng.com/blog/',
  publishTime: 1563697765000,
  time: 1563697765000,
  title: '阮一峰的网络日志',
  url: 'http://www.ruanyifeng.com/blog/atom.xml',
}

export const article: IArticle = {
  _id: 'tag:www.ruanyifeng.com,2019:/blog//1.2100',
  _rev: '5-ff6fc450616071a8f7d022b7532e85aa',
  author: 'ruanyifeng',
  categories: [],
  comments: 'article comments',
  createTime: 1563697765000,
  deleteTime: 0,
  description: 'article description',
  enclosures: [],
  feedId: 'http://www.ruanyifeng.com/blog/atom.xml',
  image: '',
  isDayFirst: true,
  isStarred: false,
  isUnread: false,
  link: 'http://www.ruanyifeng.com/blog/2019/07/weekly-issue-65.html',
  originLink: '',
  publishTime: 1563697765000,
  summary: 'article summary',
  time: 1563697765000,
  title: 'article title',
}
