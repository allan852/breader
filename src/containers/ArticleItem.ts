import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux'
import { Dispatch } from 'redux'
import ArticleItem, { IArticleItemOwnProps, IArticleItemReduxDispatch, IArticleItemReduxState } from '../components/ArticleItem'
import { IReduxAction, IReduxState } from '../schemas'

const mapStateToProps: MapStateToProps<IArticleItemReduxState, IArticleItemOwnProps, IReduxState> = (state: IReduxState) => ({
    feedsMap: state.feeds.map,
})

const mapDispatchToProps: MapDispatchToProps<IArticleItemReduxDispatch, IArticleItemOwnProps> = (dispatch: Dispatch<IReduxAction>) => ({
})

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(ArticleItem)
