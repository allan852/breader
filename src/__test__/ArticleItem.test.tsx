import Enzyme from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import EnzymeToJson from 'enzyme-to-json'
import Immutable from 'immutable'
import React from 'react'
import ArticleItem, { IArticleItemProps } from '../components/ArticleItem'
import { IFeed } from '../schemas'
import { feed, intl } from './MockData'

Enzyme.configure({ adapter: new EnzymeAdapter() })

describe('ArticleItem Testing', () => {
    const propsMock: IArticleItemProps = {
        author: 'Author',
        feedId: feed._id,
        feedsMap: Immutable.Map<IFeed>({[feed._id]: feed}),
        guid: 'guid',
        key: 1,
        summary: 'summary',
        time: 1563726737427,
        title: 'Title',
    }

    it('dom testing', () => {
        const wrapper = Enzyme.shallow<
            React.Component,
            IArticleItemProps
        >(<ArticleItem {...propsMock} />, {
            context: { intl },
        })
        expect(wrapper.props().author).toBe('Author')
    })

    it('snapshot testing', () => {
        const wrapper = Enzyme.mount(<ArticleItem {...propsMock} />, {
            context: { intl },
        })
        const tree = EnzymeToJson(wrapper)
        // expect(tree).toMatchSnapshot()
    })
})
