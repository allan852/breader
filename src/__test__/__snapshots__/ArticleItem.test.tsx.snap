// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleItem Testing snapshot testing 1`] = `
<injectIntl(ArticleItem)
  author="Author"
  feedId="http://www.ruanyifeng.com/blog/atom.xml"
  feedsMap={
    Immutable.Map {
      "http://www.ruanyifeng.com/blog/atom.xml": Object {
        "_id": "http://www.ruanyifeng.com/blog/atom.xml",
        "_rev": "5-55772bf2b4d4b74608a60ac3480334e4",
        "author": "ruanyifeng",
        "categories": Array [],
        "createTime": 1563686468437,
        "deleteTime": 0,
        "description": "Ruan YiFeng's Blog",
        "etag": "\\"11d86-58e2cc1eab031\\"",
        "favicon": "http://www.ruanyifeng.com/favicon.ico",
        "generator": "Movable Type 5.2.2 (http://www.sixapart.com/movabletype/)",
        "language": "",
        "link": "http://www.ruanyifeng.com/blog/",
        "publishTime": 1563697765000,
        "time": 1563697765000,
        "title": "阮一峰的网络日志",
        "url": "http://www.ruanyifeng.com/blog/atom.xml",
      },
    }
  }
  guid="guid"
  summary="Summary"
  time={1563726737427}
  title="Title"
>
  <ArticleItem
    author="Author"
    feedId="http://www.ruanyifeng.com/blog/atom.xml"
    feedsMap={
      Immutable.Map {
        "http://www.ruanyifeng.com/blog/atom.xml": Object {
          "_id": "http://www.ruanyifeng.com/blog/atom.xml",
          "_rev": "5-55772bf2b4d4b74608a60ac3480334e4",
          "author": "ruanyifeng",
          "categories": Array [],
          "createTime": 1563686468437,
          "deleteTime": 0,
          "description": "Ruan YiFeng's Blog",
          "etag": "\\"11d86-58e2cc1eab031\\"",
          "favicon": "http://www.ruanyifeng.com/favicon.ico",
          "generator": "Movable Type 5.2.2 (http://www.sixapart.com/movabletype/)",
          "language": "",
          "link": "http://www.ruanyifeng.com/blog/",
          "publishTime": 1563697765000,
          "time": 1563697765000,
          "title": "阮一峰的网络日志",
          "url": "http://www.ruanyifeng.com/blog/atom.xml",
        },
      }
    }
    guid="guid"
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en-US",
        "formatDate": [Function],
        "formatDateToParts": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatNumberToParts": [Function],
        "formatPlural": [Function],
        "formatRelativeTime": [Function],
        "formatTime": [Function],
        "formatTimeToParts": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralRules": [Function],
          "getRelativeTimeFormat": [Function],
        },
        "locale": "en-US",
        "messages": Object {
          "addFeed": "Add Feed",
          "allArticlesAreReadNow": "All articles are read now.",
          "delete": "Delete",
          "doYouWantSetAllArticlesBeRead": "Do you want set all articles be read?",
          "feedUrl": "Feed Url",
          "feeds": "Feeds",
          "feedsAreUpdated": "Feeds are Updated",
          "inputKeywords": "Please input the keywords",
          "invalidFeedUrl": "Invalid Feed Url",
          "keywords": "Keywords",
          "languages": "Languages",
          "menuAllItems": "All Items",
          "menuStarred": "Starred",
          "menuSubscriptions": "Subscriptions",
          "menuUnread": "Unread",
          "noFeeds": "No Feeds",
          "noMatched": "No Matched",
          "settings": "Settings",
          "somethingWrong": "Something Wrong!",
          "title": "Breader",
          "unfoundFeed": "Unfound Feed",
          "unknown": "Unknown",
        },
        "onError": [Function],
        "textComponent": Symbol(react.fragment),
        "timeZone": undefined,
      }
    }
    summary="Summary"
    time={1563726737427}
    title="Title"
  >
    <div
      className="article-item undefined"
    >
      <div
        className="item-sider"
      >
        <Avatar
          shape="square"
          size={22}
          src="http://www.ruanyifeng.com/favicon.ico"
        >
          <span
            className="ant-avatar ant-avatar-square ant-avatar-image"
            style={
              Object {
                "fontSize": 18,
                "height": 22,
                "lineHeight": "22px",
                "width": 22,
              }
            }
          >
            <img
              onError={[Function]}
              src="http://www.ruanyifeng.com/favicon.ico"
            />
          </span>
        </Avatar>
      </div>
      <div
        className="item-main"
      >
        <div
          className="item-header"
        >
          <div
            className="item-header-left"
          >
            阮一峰的网络日志
          </div>
          <div
            className="item-header-right"
          >
            00:32
          </div>
        </div>
        <div
          className="item-content"
        >
          Title
        </div>
        <div
          className="item-footer"
        >
          Summary
        </div>
      </div>
    </div>
  </ArticleItem>
</injectIntl(ArticleItem)>
`;
